### PR TITLE
Refactor FXIOS-8428 [v125] Use single name field for addresses & update SPM with latest rust-component 125.0.20240226160923

### PR DESCRIPTION
**Pull request by issammani**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8428)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18688)

## :bulb: Description
This PR (is):
- Updates address types to have a single `name` field instead of 3 `*-name` fields
- Blocked by this PR https://github.com/mozilla/application-services/pull/6118 in a-s

ℹ️ **Note:** I built the a-s change locally and all tests related to addresses are passing

Screenshot 2024-02-13 at 15 05 04

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] ~When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)~
- [ ] ~If needed I updated documentation / comments for complex code and public methods~

### CODE REVIEW

**Comment by issammani** [comment]
@nbhasin2 @jnrahme this change can only be merged after this PR https://github.com/mozilla/application-services/pull/6118 is merged, but a-s recommends to get approval before merging the upstream change.

**Comment by issammani** [comment]
cc @dimidl

**Review by nbhasin2 — APPROVED** [review]
Preliminary Approval @issammani 
However we need to wait for A~S

**Review by nbhasin2 — APPROVED** [review]

**Comment by nbhasin2** [comment]
This is required for https://github.com/mozilla-mobile/firefox-ios/pull/18885

Going to backport

**Comment by nbhasin2** [comment]
@mergify backport release/v124

**Pull request merged**
